sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptions.java

```diff
@@ -322,6 +323,13 @@ public String create(PipelineOptions options) {
   }
 
   /**
+   * Return a map of properties which correspond to {@link RuntimeValueProvider},
+   * keyed by the property name.  The value is a map containing type and default
+   * information.
+   */
+  Map<String, Map<String, Object>> outputRuntimeOptions();
```

Comment:
Return value: The intention was to allow to write a side file (or something else, like send it to an API) with metadata about pipeline options, for validation and other magic.  I'm open to suggestions about return value, but it seems like anything tighter would disallow some of the uses that already exist. 

Naming: I might be misattributing this, but it appears that if I call it get*, then it runs afoul of the Bean property magic that currently makes PipelineOptions work.